libr/bin/format/elf/elf.c

```diff
@@ -2718,28 +2719,32 @@ static size_t populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *rel
 		size = get_size_rel_mode (rel_mode);
 
 		for (j = get_next_not_analysed_offset (bin, bin->g_sections[i].offset, 0, bin->baddr);
-			j < bin->g_sections[i].size;
+			j < bin->g_sections[i].size && p < num_relocs;
 			j = get_next_not_analysed_offset (bin, bin->g_sections[i].offset, j + size, bin->baddr)) {
-
-			if (!read_reloc (bin, relocs + pos, rel_mode, bin->g_sections[i].offset + j)) {
+			if (!j) {
 				break;
 			}
 
-			fix_rva_and_offset (bin, relocs + pos, i);
-			pos++;
+			if (!read_reloc (bin, relocs + p, rel_mode, bin->g_sections[i].offset + j)) {
+				break;
+			}
+			if (p >= num_relocs) {
```

Comment:
Why? I think you already checked this in the for condition

Reply:
this is exactly what makes relocs not spit out trash

Reply:
no.... this is already done in the for loop condition (`p < num_relocs`) so this is not necessary.